components/camel-debezium/pom.xml

```diff
@@ -0,0 +1,100 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one or more
+    contributor license agreements.  See the NOTICE file distributed with
+    this work for additional information regarding copyright ownership.
+    The ASF licenses this file to You under the Apache License, Version 2.0
+    (the "License"); you may not use this file except in compliance with
+    the License.  You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.apache.camel</groupId>
+        <artifactId>components</artifactId>
+        <version>3.0.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>camel-debezium</artifactId>
+    <packaging>jar</packaging>
+    <name>Camel :: Debezium</name>
+    <description>Camel Debezium support</description>
+
+    <properties>
+    </properties>
+
+    <dependencies>
+
+        <!-- camel -->
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-support</artifactId>
+        </dependency>
+
+        <!-- debezium embedded engine -->
+        <dependency>
+            <groupId>io.debezium</groupId>
+            <artifactId>debezium-embedded</artifactId>
+            <version>${debezium-version}</version>
+        </dependency>
+
+        <!-- debezium connectors -->
+        <dependency>
```

Comment:
Instead of relying on a specific connector, I'd suggest to add their versions to your BOM (if there is such thing in Camel) and then have users depend on the specific connector they want.

Reply:
Can you please clarify further what do you mean here? I am not really sure if that is possible in camel but maybe I am missing something :/

Reply:
@omarsmak, what I mean is this: a user may want to use the DBZ Postgres connector instead of the one for MySQL. This dependeny would be wrong in this case.

Reply:
Yeah that is the draw back of having the whole debezium component that includes all connectors. However, I thought of separating this component to multiple components that reflect each connect, for example like this: 
`camel-debezium-core` -> covers the core for the component.
`camel-debezium-mysql` -> only for mysql connector. 
`camel-debezium-mongodb` -> only for mongodb connector. 
.. 
..
However, the initial version for the component kept only for mysql, when we add the other connectors, we can use the above structure of you think it would make more sense.

Reply:
We need to check the license of mysql. We are not suppose to ship the GPL or LGPL third part bundle in Apache release.

Reply:
The debezium connector mysql uses mysql-binlog-connector-java which is ASF 2 licensed, this connector uses in turn mysql-connector-java which is GPL 2.0

Reply:
This is the dependency tree

[INFO] +- io.debezium:debezium-connector-mysql:jar:0.9.5.Final:compile
[INFO] |  +- io.debezium:debezium-ddl-parser:jar:0.9.5.Final:compile
[INFO] |  |  \- org.antlr:antlr4-runtime:jar:4.7:compile
[INFO] |  +- com.github.shyiko:mysql-binlog-connector-java:jar:0.19.1:compile
[INFO] |  \- mysql:mysql-connector-java:jar:8.0.13:compile
[INFO] |     \- com.google.protobuf:protobuf-java:jar:3.6.1:compile

Reply:
To give some context, this the presence of mysql-connector-java in all the Apache organization POM

https://github.com/search?l=Maven+POM&q=org%3Aapache+%27mysql-connector-java%27&type=Code

@davsclaus @WillemJiang

Reply:
one thing we could do is: exclude the mysql-binlog-connector-java, add it explicitly with the exclusion of mysql-connector-java and add mysql-connector-java as provided dependency

Reply:
Yeah this is a problem, the GPLv2 license is X, and we should remove it. Also as @gunnarmorling have mentioned then I dont see that the camel-debezium component should be tied to MySQL as debezium works with other databases. So this component should IMHO be designed as neutral and not tied to MySQL at all. For unit testing the camel-debezium its a different matter, its okay to do testing with MySQL and GPLv2 licensed libraries.

Reply:
Can't we, for the moment, use the workaround of the provided dependency?

Reply:
maybe it would be sufficient to exclude the mysql-connector-java from the binlog dependency and add a note in the documentation

Reply:
@davsclaus sure we need to make it neutral but the user still needs to add the dependency for the desired the connector via a camel component in order to parse the specific configurations during the compile time for camel docs and property configurations. We have this ticket open to make this as easy as possible here: https://issues.apache.org/jira/browse/CAMEL-14009. Me personally I am routing to exclude `mysql-connector-java` and let the user add it externally.

Reply:
Okay so is `mysql-connector-java` only used for some documentation generation? If so then CAMEL-14009 should make it so that this JAR is NOT in use at all when end users are using Camel. However its okay if its part of some kind of build time maven plugin or something. 

However at runtime for using Camel then this JAR must not be present at all. Also if that means the end user must add it manually as a Camel component should be usable out of the box as-is and not hit license problems with GPLv2 libraries etc.

Reply:
Yes, theoretically CAMEL-14009 should somehow enable that (haven't validate it yet), but as you mentioned, these dependencies will be only needed for maven build in order to generate the concrete classes. IMHO at the mean time, we can exclude the tricky `mysql-connector-java` dependency until CAMEL-14009 is ready which I am planning to work on soon, if everyone is okay with that of course.

Reply:
Test fails

[ERROR] Tests run: 5, Failures: 0, Errors: 3, Skipped: 0, Time elapsed: 0.002 s <<< FAILURE! - in org.apache.camel.component.debezium.configuration.MySqlConnectorEmbeddedDebeziumConfigurationTest
[ERROR] testIfCreatesConfig(org.apache.camel.component.debezium.configuration.MySqlConnectorEmbeddedDebeziumConfigurationTest)  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: com/mysql/cj/jdbc/Driver
	at org.apache.camel.component.debezium.configuration.MySqlConnectorEmbeddedDebeziumConfigurationTest.testIfCreatesConfig(MySqlConnectorEmbeddedDebeziumConfigurationTest.java:38)
Caused by: java.lang.ClassNotFoundException: com.mysql.cj.jdbc.Driver
	at org.apache.camel.component.debezium.configuration.MySqlConnectorEmbeddedDebeziumConfigurationTest.testIfCreatesConfig(MySqlConnectorEmbeddedDebeziumConfigurationTest.java:38)

[ERROR] testIfCreatesConfig(org.apache.camel.component.debezium.configuration.MySqlConnectorEmbeddedDebeziumConfigurationTest)  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class io.debezium.connector.mysql.MySqlConnectorConfig
	at org.apache.camel.component.debezium.configuration.MySqlConnectorEmbeddedDebeziumConfigurationTest.testIfCreatesConfig(MySqlConnectorEmbeddedDebeziumConfigurationTest.java:38)

[ERROR] testIfCreatesConfig(org.apache.camel.component.debezium.configuration.MySqlConnectorEmbeddedDebeziumConfigurationTest)  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class io.debezium.connector.mysql.MySqlConnectorConfig
	at org.apache.camel.component.debezium.configuration.MySqlConnectorEmbeddedDebeziumConfigurationTest.testIfCreatesConfig(MySqlConnectorEmbeddedDebeziumConfigurationTest.java:38)

Reply:
without the mysql-connector-java

Reply:
but we can add it as test dependencies

Reply:
How about change the scope to be provided? In this way, we don't ship the mysql connector in our bundle release.

Reply:
Possible attempt

https://github.com/apache/camel/pull/3191